airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -536,55 +534,28 @@ def streams(self, config: MutableMapping[str, Any]) -> List[Stream]:
             ),
         ]
 
-        state_manager = ConnectorStateManager(state=self._state)
+        state_manager = ConnectorStateManager(stream_instance_map={s.name: s for s in streams}, state=self._state)
+        converter = EpochValueConcurrentStreamStateConverter()
+        start = datetime.fromtimestamp(self._start_date_to_timestamp(config), timezone.utc)
+        slice_range = timedelta(days=config["slice_range"])
+
         return [
-            self._to_concurrent(
+            self._convert_to_concurrent_stream(
+                logger,
                 stream,
-                datetime.fromtimestamp(self._start_date_to_timestamp(config), timezone.utc),
-                timedelta(days=config["slice_range"]),
-                state_manager,
+                cursor=self.initialize_cursor(
+                    stream,
+                    state_manager,
+                    converter,
+                    self._SLICE_BOUNDARY_FIELDS_BY_IMPLEMENTATION.get(type(stream)),
+                    start,
+                    converter.get_end_provider(),
+                    slice_range=slice_range,
+                ),
             )
             for stream in streams
         ]
 
-    def _to_concurrent(
-        self, stream: Stream, fallback_start: datetime, slice_range: timedelta, state_manager: ConnectorStateManager
-    ) -> Stream:
-        if stream.name in self._streams_configured_as_full_refresh:
-            return StreamFacade.create_from_stream(
-                stream,
-                self,
-                entrypoint_logger,
-                self._create_empty_state(),
-                FinalStateCursor(stream_name=stream.name, stream_namespace=stream.namespace, message_repository=self.message_repository),
-            )
-
-        state = state_manager.get_stream_state(stream.name, stream.namespace)
-        slice_boundary_fields = self._SLICE_BOUNDARY_FIELDS_BY_IMPLEMENTATION.get(type(stream))
-        if slice_boundary_fields:
-            cursor_field = CursorField(stream.cursor_field) if isinstance(stream.cursor_field, str) else CursorField(stream.cursor_field[0])
```

Comment:
this was the behavior i commented in the CDK method. I kind of feel like this truncation getting swallowed is not the best pattern.

@girarda its probably been a while, but do you recall a specific reason why we didn't want to throw an error here? or can we be more explicit

Reply:
I doubt there's a real reason. I'm onboard with adding a validation that the cursor field isn't composite 👍